guides/github.md

```diff
@@ -0,0 +1,130 @@
+# How to secure GitHub repositories
+
+It is recommended to turn on 2FA for most repositories, especially if building
+a package that is a dependency of multiple other packages.
+
+At least for critical packages, administrators should enable branch
+protection. Require CI to pass before merging to the main branch. Allow only
+repository owners to merge PRs.
+
+> [!WARNING]
+> It is recommended to run workflows only after the PR has been reviewed. The
+> "Require approval for all outside collaborators" setting is the recommended
+> one. However, this can cause friction in repositories with a high number of
+> PRs but small number of contributors. In this case, if all contributors are
+> trusted, "Require approval for first-time contributors" is a valid option,
+> but dedicate more efforts to separate sensitive steps into workflows that
+> can only run after approval.
+
+It is ideal to require PR reviews before merging. For security critical
+packages, having at least two reviewers is ideal, as this alleviates the risk
+of sock-puppet accounts.
+
+To keep dependencies up to date, it is advisable to enable
+[Dependabot][dependabot] or [Renovate][renovate] or a similar tool which will
+create PRs to update dependencies and alert on vulnerabilities in
+dependencies. Both of these tools can be configured to send a PR at regular
+intervals (e.g., once a week).
+
+It is recommended to install [Scorecards action][scorecard] for public
+repositories and trying to improve the score as high as possible. This is a
+scanner for security best practices, most of which are already discussed in
+this document.
+
+## Securing GitHub Actions
+
+It is preferable to use the hosted runners as possible. The large runners
+supported by GitHub come with usage costs, but are better from the point of
+view of supply chain security than hosting your own infrastructure for CI.
```

Comment:
What is the reasoning for this claim?

Reply:
You no longer need to take steps in securing your own infrastructure. It's a trade-off of costs between what you'd spend on securing your infra vs what you'd spend on paying GitHub

Reply:
This sounds like a rather bold assumption that Github has more secure infrastructure than some machine on managed amazon cloud.

Reply:
It's more an assumption on the fact that GitHub has a security team that spent a lot of time to look at securing infrastructure. Of course if someone spends time to secure their own infra to what they think is a better security standard, they should use their own infra.

A student that just starts a new project? Better to use Github's CI that their own. A big company with dedicated security team and cloud infra? Better to use their own CI